search_service/proxy/base.py

```diff
@@ -45,11 +45,12 @@ def delete_document(self, *,
         pass
 
     @abstractmethod
-    def fetch_table_search_results_with_filter(self, *,
-                                               query_term: str,
-                                               search_request: dict,
-                                               page_index: int = 0,
-                                               index: str = '') -> SearchTableResult:
+    def fetch_search_results_with_filter(self, *,
+                                         query_term: str,
+                                         search_request: dict,
```

Comment:
Do we validate if the key is filterable key? For example, if client sends some unknown key, how search service would react?

example: `{  'bogus': ['mode'] }`

Reply:
currently we validate the search filter value inside the proxy https://github.com/lyft/amundsensearchlibrary/blob/master/search_service/proxy/elasticsearch.py#L365